src/Layouts/stars/feed/addComment.tsx

```diff
@@ -0,0 +1,47 @@
+import { TextField, Avatar, Button } from "@mui/material";
+import { noteType } from "../../../assets/star";
+import { useState } from "react";
+
+interface commentProps {
+  replyTo: noteType | undefined;
+  setReplyTo: (param: noteType | undefined) => void;
+  addNote: (note: noteType) => void;
+}
+
+const AddComment = ({ replyTo, setReplyTo, addNote }: commentProps) => {
+  const [input, setInput] = useState<string>("");
+
+  const addComment = () => {
+    let newNote: noteType = {
+      id: Math.random(),
+      publisher: "מיקי - מאב",
+      note: input,
+      time: new Date(),
+      repliesTo: replyTo?.id,
+    };
+    
+    setInput("");
+    setReplyTo(undefined);
+    addNote(newNote);
+  };
+
+  return (
+    <div className="addComment">
+      <Avatar id="avatar" alt="Remy Sharp" src="/static/images/avatar/2.jpg" />
+      <TextField
```

Comment:
i'm pretty sure that if you will do it as a [uncontrolled  ](https://mui.com/components/text-fields/#uncontrolled-vs-controlled) component you will have the option not to save the value as a state and only submit (mabey as a form) at the end  ( i think its preferred  because  you don't render the component endlessly

Reply:
could not find any other solution except for ref wich did not work for me.
the problem with giving up the state is that there is no way to get the TextField's value except for state or ref, when clicking the button.
it would have been possible if I were calling addComment() from within the TextField itself

Reply:
tried many stuff, nothing worked.
google says the only alternative is to use refs, it doesnt work and anyway people suggest to use state instead as it is more stable